### ISSUE TITLE
DB editor does not add new classes if they are empty

### ISSUE THREAD

**Issue by jkiviluo** [spine-tools/Spine-Toolbox#2312]
Commit will not add new empty (multi-dimensional?) classes. Apparently there needs to be some data there.

**Comment by jkiviluo** [comment]
With data included, I got this error (and commit apparently failed):

```
DBAPIError while inserting entity_class items: ('UNIQUE constraint failed: entity_class_dimension.entity_class_id, entity_class_dimension.dimension_id, entity_class_dimension.position',)
```

**Comment by PiispaH** [comment]
Might be related to #2293

**Comment by jkiviluo** [comment]
Right. And it wasn't actually committed. I closed and opened the DB editor, but that's not enough (apparently stuff remains in the memory). Only closing and starting Toolbox revealed that new entity class and it's data were not in the DB.

**Comment by manuelma** [comment]
What are steps to reproduce this? I just added and committed an empty multi-D class with no problem?

**Comment by jkiviluo** [comment]
It works for me too, if I start from an empty DB. But fails on the attached one. Of course, maybe it's due to corrupt DB that wouldn't happen anymore, so not sure if should be debugged (and I would just need to reimplement my DB).

[generic_energy_data_specification.sqlite.txt](https://github.com/spine-tools/Spine-Toolbox/files/12610248/generic_energy_data_specification.sqlite.txt)

**Comment by jkiviluo** [comment]
I tried to look into this, since it's stopping me from progressing. It seems like the referential integrity between entity_class and entity_class_dimension tables is not working. I have rows in entity_class_dimension that have entity_class_ids not found in entity_class (since the database has seen deletion of entity_classes). This causes the error message when trying to insert records that would re-use the entity_class_id numbers.

I don't know where to fix this... Probably somewhere in the database definition, but I leave it to you @manuelma. I'll try to fix my database manually meanwhile.

**Comment by manuelma** [comment]
My hope is @PiispaH already fixed this with https://github.com/spine-tools/Spine-Database-API/commit/7fa4b1e3549cf77387f96e7dfe735e733e681cff

It is somewhat clear to me how things would get messy the way you @jkiviluo describe, prior to the changes introduced by @PiispaH in the above commit.

Should we try and save the corrupted DBs? there might be a way...

**Comment by jkiviluo** [comment]
No need on my part - I've already remade the DB. So, unless someone requests that, let's not waste time trying to do that. Presumably there are so many ways things can go wrong that it can be futile.
